src/me/remie/vulcan/leaguetasks/utils/LeaguePanel.java

```diff
@@ -0,0 +1,123 @@
+package me.remie.vulcan.leaguetasks.utils;
+
+import simple.hooks.simplebot.Game;
+import simple.hooks.wrappers.SimpleWidget;
+import simple.robot.api.ClientContext;
+
+public class LeaguePanel {
+    private ClientContext ctx;
+    private static final int WIDGET_ID = 657;
+
+    public LeaguePanel(final ClientContext ctx) {
+        this.ctx = ctx;
+    }
+
+    public boolean doWeNeedToClearFilters() {
+        final SimpleWidget tierDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_TIER.getWidgetId());
+        if (!isOptionSelected(tierDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget typeDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_TYPE.getWidgetId());
+        if (!isOptionSelected(typeDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget areaDropDownMenu = ctx.widgets.getWidget(WIDGET_ID,LeaguePanelFilterType.FILTER_AREA.getWidgetId());
+        if (!isOptionSelected(areaDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget completedDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_COMPLETED.getWidgetId());
+        if (!isOptionSelected(completedDropDownMenu)) {
+            return true;
+        }
+
+        return false;
```

Comment:
```suggestion
for (LeaguePanelFilterType filterType : LeaguePanelFilterType.values()) {
            SimpleWidget dropDownMenu = ctx.widgets.getWidget(WIDGET_ID, filterType.getWidgetId());
            if (!isOptionSelected(dropDownMenu)) {
                return true;
            }
        }
        return false;
```
Good step in the right direction but we can reduce code duplication even more!